### ISSUE TITLE
Plugin directory possible sql improvements

### ISSUE THREAD

**Issue by oguzkocer** [wordpress-mobile/WordPress-FluxC-Android#682]
We need to look into a more efficient way to fetch/insert wporg plugins and plugin directory models. Since it's not a priority and I have limited time to finish the v1, I am skipping this optimization, but we should definitely go back to it and make some improvements. We could probably utilize a foreign key in the `PluginDirectoryModelTable` pointing to the `WPOrgPluginModelTable`.

Also, since there is a disconnect between the `PluginDirectoryModelTable` and `WPOrgPluginModelTable`, there is a possibility that some of the wporg plugins might be fetched once and never get deleted since we never remove the wporg plugins from the DB. We could use a pruning process once in a while to maybe delete the whole `WPOrgPluginModelTable` or remove the ones that are not used for a while.

We should also look into using a DB query for the `PluginSqlUtils.getLastRequestedPageForDirectoryType`. I've tried several ways to do this with WellSql and was unable to find a way to get it to work. We could use a custom sql, but I really don't want to do it just because it's more error prone and this method's usage is very rare.

**Comment by oguzkocer** [comment]
We won't be working on this issue right now, so I am unassigning myself but keeping the issue open. I think I am mostly OK with the things being the way they are. I'll most likely need help to figure these out and I don't want to block people (or myself) because these optimizations will not a big difference right now.

**Comment by oguzkocer** [comment]
- [ ] We should convert the slug fields in both plugin models to unique and add an index for it.

**Comment by oguzkocer** [comment]
Pruning issue has been separated into its own issue #720. With the changes in #717 & #718 we have been able to do a lot of improvements to the sql queries and we'll continue to do so as we feel necessary. At least for now, the performance issue I was experiencing seems to have been resolved and I am mostly happy with the queries with where they are. Any other improvements we make will probably be included in other issues, so we don't need to track this anymore.
